crates/compiler/gen_dev/src/generic64/x86_64.rs

```diff
@@ -1396,6 +1439,46 @@ fn addss_freg32_freg32(buf: &mut Vec<'_, u8>, dst: X86_64FloatReg, src: X86_64Fl
     }
 }
 
+/// `MULSD xmm1,xmm2/m64` -> Multiply the low double-precision floating-point value from xmm2/mem to xmm1 and store the result in xmm1.
+#[inline(always)]
+fn mulsd_freg64_freg64(buf: &mut Vec<'_, u8>, dst: X86_64FloatReg, src: X86_64FloatReg) {
+    let dst_high = dst as u8 > 7;
+    let dst_mod = dst as u8 % 8;
+    let src_high = src as u8 > 7;
+    let src_mod = src as u8 % 8;
```

Comment:
this is copied from the addition functions. I'm assuming that is correct because I don't fully understand what is going on here

Reply:
Look at the end of file and make some tests. They use a disassembler for verification. If that passes, this should all be good. We probably should add a helper at some point similar to `binop_reg64_reg64`.